### PR TITLE
Setup Database

### PR DESCRIPTION
**Pull request by n6rayan**
**Ticket**: [Setup Database](https://trello.com/c/yZDlfDM5)

This review includes the following changes:

- Updates to documentation
- Config file for development environment
- Database migration files
- Connection module for database
- Modules to perform CRUD actions on job, client and schema db tables
- Test coverage for those actions

### CODE REVIEW

**Comment by n6rayan** [comment]
[Setup Database](https://trello.com/c/yZDlfDM5/4-setup-database)

File: src/db/job/job.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+const fetchJob = (jobID, apiKey) => {
+    return new Promise((resolve, reject) => {
+        db('job')
+        .where({ id: jobID, api_key: apiKey })
+        .then(result => resolve(result))
+        .catch(error => reject(error));
+    });
+};
+
+module.exports.createJob = createJob;
```
Comment (markogrady1) [inline src/db/job/job.js]: definity not a fan of this pattern. To much duplication.
I suggest adding all functions to the `job` object and export that. 
Any functions you want private stay outside of object.

`let jobs = {};

jobs.createJob = function() {
  ......
}

jobs.deleteJob = function() {
  ......
}

module.exports = jobs;`

File: src/db/job/job.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+const db = require('../connection');
```
Comment (markogrady1) [inline src/db/job/job.js]: The job, client, schema modules are near identical apart from the query and table name.
You could have one DB module that has functions **create, delete, update and fetch** with the arguments passed in telling it what table and query should be used.

Then these modules can be used to build those arguments. 

i.e 
```
db.insert('client', {job: jobData,  api_key: apiKey});
```

Comment (n6rayan) [reply src/db/job/job.js]: Updated to this now.

File: src/db/job/job.js (hunk 1 above)
Comment (n6rayan) [reply src/db/job/job.js]: Updated!

**Pull request merged**